src/Controller/FoodHeroController.php

```diff
@@ -126,12 +129,67 @@ public function showOffer(FoodHero $foodhero, Offer $offer)
      * @param Offer $offer
      * @return Response
      */
-    public function acceptOffer(FoodHero $foodhero, Offer $offer)
+    public function acceptOffer(FoodHero $foodhero, Offer $offer) : Response
     {
         $em = $this->getDoctrine()->getManager();
         $offer->setFoodhero($foodhero);
         $em->flush();
         
         return $this->redirectToRoute('foodhero_list_offers', ['id' => $foodhero->getId()]);
     }
+
+
+    /**
+     * @Route("/position/{latitude}/{longitude}")
+     * @param float $latitude
+     * @param float $longitude
+     * @param SessionInterface $session
+     * @return Response
+     */
+    public function showCoordinates(?float $latitude, ?float $longitude, SessionInterface $session)  : Response
+    {
+
+            $session->set('latitude', $latitude);
+            $session->set('longitude', $longitude);
+            return $this->json('');
+    }
+
+    /**
+     * @Route("/{foodhero}/oneOffer/{offer}", name="foodhero_offer_card")
+     * @return Response
+     * @throws \Exception
+     */
+    public function showOneOffer(
+        FoodHero $foodhero,
+        Offer $offer,
+        DistanceCalculator $distanceCalculator,
+        SessionInterface $session
+    ): Response {
+
+        if ($session!=null) {
+            $company = $offer->getCompany();
+            $association=$offer->getAssociation();
+            $distanceAssoComp = $distanceCalculator->calculateDistanceFromAdresses($company, $association);
+            $distance = $distanceCalculator->calculateDistanceFromGps(
+                $session->get('latitude'),
+                $session->get('longitude'),
+                $company
+            );
+            $distanceTotal=$distance+$distanceAssoComp;
+
+            return $this->render('Visitor/FoodHero/showCard.html.twig', [
+                'foodHero' => $foodhero,
+                'distance' => $distance,
+                'distanceTotal'=>$distanceTotal,
+                'offer' => $offer
+            ]);
+        } else {
```

Comment:
non pour ton else, tu vois bien que tu répète 2 fois le meme code dans ton this->render. Tu peux mettre ton if ($session->has('latitude')) juste avant le $distance (ligne 173 et tu le ferme ligne 179). et dans ton return tu fais un 
$distance ?? 'non calculée' et idem pour distanceTotal (ou tu gères ça un peu différement, tu renvois null si pas de valeur et tu gères l'affichage de la chainte "non calculée" uniquement dans la vue.